### PR TITLE
Feature/high contrast

### PR DESCRIPTION
**Pull request by aoneill01**
### Resolves

- Resolves [ENA-304](https://scratchfoundation.atlassian.net/browse/ENA-304)

### Proposed Changes

- Use updated `scratch-gui` to pull in high contrast changes
- Update some elements (primarily About and Privacy Policy dialogs) to use new purple theme
- Add modal to select USB device during micro:bit firmware update process
  - In the online version this is provided by the browser. Electron does not supply this automatically, so we need to tie in to [their api](https://www.electronjs.org/docs/latest/tutorial/devices#webusb-api).
- Update Electron
  - Electron was on an old-enough version of Chrome that the CSS used for some of the hover effects in the new Settings menu was not working.

### Reason for Changes

The online version of Scratch received a significant style update for high contrast accessibility. We want to include those changes in the offline apps, including this project.

### Test Coverage

Manual testing on both Windows and Mac.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,8 +42,8 @@
     "babel-loader": "^8.1.0",
     "babel-plugin-react-intl": "^7.5.7",
     "copy-webpack-plugin": "^5.1.1",
-    "electron": "^15.3.1",
-    "electron-builder": "^22.13.1",
+    "electron": "^25.2.0",
+    "electron-builder": "^24.4.0",
```
Comment (cwillisf) [inline package.json:46]: 😎
(😅)

File: src/main/index.js [hunk 2]
```diff
@@ -230,6 +230,52 @@ const createPrivacyWindow = () => {
...
+        height: 300,
+        parent: _windows.main,
+        search: 'route=usb',
+        modal: true,
+        frame: false
+    });
+
+    // Filters from navigator.usb.requestDevice do not appear to be available here.
+    // Hard code to micro:bit since that is the only device that currently uses this api.
+    const getIsMicroBit = device => device.vendorId === 0x0d28 && device.productId === 0x0204;
```
Comment (cwillisf) [inline src/main/index.js:245]: I think this is fine for now, but do you know how we'd handle this if we supported more than one kind of device? It didn't seem obvious when I skimmed the API docs you linked. Or, given that the vendor and product ID are provided to `navigator.usb.requestDevice`, I don't suppose devices are getting filtered before the `usb-device-added` event...?

**Review by cwillisf — COMMENTED** [review]
Looks good so far! Looks like you might need scratchfoundation/scratch-gui#9105 for the Windows build. Not sure what's up with the Mac build, though...

File: src/renderer/usb.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    };
+
+    const deviceHandler = deviceId => () => {
+        ipcRenderer.send('usb-device-selected', deviceId);
+        setSelectedDeviceId(null);
+    };
+
+    return (
+        <main>
+            Select your USB device:
```
Comment (apple502j) [inline src/renderer/usb.jsx:34]: (Day 1665 of asking for fully translated Scratch App...)

File: src/main/index.js (hunk 2 above)
Comment (aoneill01) [reply src/main/index.js:245]: No, the devices are not getting filtered. Without this I was seeing all USB devices connected to my computer. I also tried inspecting the event passed to the `'select-usb-device'` handler hoping the filter contents would be in there, but I didn't see it. 

If we really needed to support more than one kind of device, the only thing that comes to mind is somehow having gui communicate that to Electron using Electron APIs. I don't really like that.

When I have some time I'll see if I can find a place to post this question to Electron. Maybe they will add it if it isn't something I'm missing.

**Review by cwillisf — APPROVED** [review]
Looks good! I'm glad the `--max-old-space-size` thing worked.

**Pull request merged**